src/charm_state.py

```diff
@@ -5,32 +5,63 @@
 
 import dataclasses
 import logging
+import re
+from typing import Optional
 
 from ops import CharmBase
 
 logger = logging.getLogger(__name__)
 
 COS_AGENT_INTEGRATION_NAME = "cos-agent"
+HOSTNAME_PORT_PATTERN = re.compile(r"^[\w\-.]+:\d+$")
+
+
+class CharmConfigInvalidError(Exception):
+    """Raised when charm config is invalid.
+
+    Attributes:
+        msg: Explanation of the error.
+    """
+
+    def __init__(self, msg: str):
+        """Initialize a new instance of the CharmConfigInvalidError exception.
+
+        Args:
+            msg: Explanation of the error.
+        """
+        self.msg = msg
 
 
 @dataclasses.dataclass(frozen=True)
 class State:
     """The charm state.
 
     Attributes:
-        proxy_config: Proxy configuration.
-        _charm: The charm instance.
+        is_metrics_logging_available: Whether the charm is able to issue metrics.
+        aproxy_proxy: The socket address of the proxy to configure aproxy with.
```

Comment:
This name sounds a bit odd to me. How about `proxy_address`?

Reply:
I want to include "aproxy" to avoid confusing it with the other proxy settings from juju (e.g. from `JUJU_CHARM_HTTP_PROXY`). Maybe `aproxy_address`?